scripts/collect_api_version_for_multi_api_sdk/main.py

```diff
@@ -0,0 +1,200 @@
+                self.find_versions_from_json(provider, version, self.multi_api_version_from_profiles)
+
+    def get_api_version_from_rest_api_profile(self):
+        if not self.provider_mapping_package:
+            self.get_multiapi_from_rest_api()
+        # map package name to api version like {'azure-mgmt-msi': {'2018-11-30'}}
+        url_path = SOURCE_FILE['rest-api-profile']
+        git_path = url_path.split('main/')[-1]
+        file_paths = self.rest_repo.get_contents(git_path)
+        for file in file_paths:
+            if '.json' not in file.name:
+                continue
+            file_name, file_contents = file.name.replace('.json', ''), file.decoded_content
+            content_dict = json.loads(file_contents.decode())
+            resource_manager = 'resource-manager' if content_dict.get('resource-manager') else 'resourcemanager'
+            for provider, version in content_dict[resource_manager].items():
+                self.find_versions_from_json(provider, version, self.multi_api_version_from_profile)
+
+    @staticmethod
+    def extract_api_version(api_version_info: Any) -> Set[str]:
+        # convert to string
+        if isinstance(api_version_info, str):
+            api_version = api_version_info
+        else:
+            api_version = api_version_info.default_api_version + str(api_version_info.profile)
+
+        return set(re.findall('\d{4}-\d{2}-\d{2}[-a-z]*', api_version))
+
+    @staticmethod
+    def write_file(file_name, content):
+        json_out = {k: sorted(content[k], reverse=True) for k in content}
+        with open(file_name, 'w') as file_out:
+            json.dump(json_out, file_out, indent=4)
+
+    def output(self):
+        # eg: change them like azure.mgmt.resource to azure-mgmt-resource
+        self.package_api_version = {k.replace('.', '-'): v for k, v in self.package_api_version.items()}
+        # output service and api version from cli
+        self.write_file('package_api_version_from_cli.json', self.package_api_version)
+        # output service and api version from profiles
+        self.write_file('package_api_version_from_profiles.json', self.multi_api_version_from_profiles)
+        # output service and api version from profile
+        self.write_file('package_api_version_from_profile.json', self.multi_api_version_from_profile)
+
+        # merge multi_api_version_from_profiles to package_api_version
+        for k, v in self.multi_api_version_from_profiles.items():
+            self.package_api_version[k] = self.package_api_version[k] | v if self.package_api_version.get(k) else v
+        # merge multi_api_version_from_profile to package_api_version
+        for k, v in self.multi_api_version_from_profile.items():
+            self.package_api_version[k] = self.package_api_version[k] | v if self.package_api_version.get(k) else v
+        # output all services and api versions
+        self.write_file('package_api_version_all.json', self.package_api_version)
```

Comment:
we could define a dict {info_dict_name: output_file_name} like:
```
{
    “package_api_version”:  "package_api_version_all.json"
    ...
}
```
then:
```
for k,v in dict:
    self.write_file(v, getattr(self, k))
```

for merged files,
```
all = {}
for k in dict.keys():
    for item in k.items():
       ....
self.write(all_files, all)
```
the up structure could make code cleaner to easy to add or delete